pkg/diff/chunk.go

```diff
@@ -221,170 +205,57 @@ func (s *randomSpliter) split(table *TableInstance, columns []*model.ColumnInfo,
-				log.Info("no data found", zap.String("table", dbutil.TableName(schema, table)), zap.String("range", limitRange), zap.Reflect("args", args))
-				return append(chunks, chunk), nil
-			}
 			return nil, errors.Trace(err)
 		}
 
-		symbolMin = gte
-		symbolMax = lte
-	}
-
-	splitValues := make([]string, 0, count)
-	valueCounts := make([]int, 0, count)
-
-	// get random value as split value
-	randomValues, randomValueCount, err := dbutil.GetRandomValues(context.Background(), db, schema, table, splitCol, count-1, limitRange, utils.StringsToInterfaces(args), s.collation)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	log.Debug("get split values by random values", zap.Stringer("chunk", chunk), zap.Reflect("random values", randomValues))
-
-	/*
-		for examples:
-		the result of GetRandomValues is:
-		mysql> SELECT `id`, count(*) count FROM (SELECT `id` FROM `test`.`test` ORDER BY RAND() LIMIT 100) rand_tmp GROUP BY `id` ORDER BY `id`;
-		+------+-------+
-		| id   | count |
-		+------+-------+
-		|    1 |     1 |
-		|    2 |     1 |
-		|    3 |    96 |
-		|    4 |     1 |
-		|    5 |     1 |
-		+------+-------+
-
-		We can assume that the 96% of this table's data is in range [id = 3], so we should use another column to split range `id = 3`,
-		just like [id = 3 AND cid > 10], [id = 3 AND cid >= 5 AND cid <= 10], [id = 3 AND cid < 5]...
-	*/
-
-	if len(randomValues) > 0 && randomValues[0] == min {
-		splitValues = append(splitValues, randomValues...)
-		valueCounts = append(valueCounts, randomValueCount...)
-		valueCounts[0]++
-	} else {
-		splitValues = append(append(splitValues, min), randomValues...)
-		valueCounts = append(append(valueCounts, 1), randomValueCount...)
-	}
-
-	if len(randomValues) > 0 && randomValues[len(randomValues)-1] == max {
-		valueCounts[len(valueCounts)-1]++
-	} else {
-		splitValues = append(splitValues, max)
-		valueCounts = append(valueCounts, 1)
+		log.Debug("get split values by random", zap.Stringer("chunk", chunk), zap.String("column", column.Name.O), zap.Reflect("random values", randomValues[i]))
 	}
 
-	/*
-		for example:
-		the splitCol is `a`;
-		the splitValues is [1, 2, 3, 4, 5];
-		the splitCounts is [1, 3, 1, 1, 1];
-
-		this means you get 3 times value 2 by random, we can assume that there amolst be a lot of rows with value 2,
-		so we need use another column `b` to split the chunk [`a` = 2] to 3 chunks.
-
-		and then the splitCol is `b`;
-		the splitValues is ['w', 'x', 'y', 'z'];
-		the splitValues is [1, 1, 1, 1];
-		the chunk [`a` = 2] will split to [`a` = 2 AND `b` < 'x'], [`a` = 2 AND `b` >= 'x' AND `b` < 'y'] and [`a` = 2 AND `b` >= 'y']
-	*/
-
-	lowerSymbol := symbolMin
-	upperSymbol := lt
-
-	for i := 0; i < len(splitValues); i++ {
-		if i == 0 && useNewColumn {
-			// create chunk less than min
-			newChunk := chunk.copyAndUpdate(splitCol, "", "", splitValues[i], lt)
-			chunks = append(chunks, newChunk)
-		}
+	for i := 0; i <= len(randomValues[0]); i++ {
```

Comment:
may be don't have same num values, because get random value for every column one by one, and the data may be changed. but we define a limit for diff: don't support online check. And I still add a new function `minLenInSlices` to get the min length in `[][]string` in 5bcf5bc95eb4091e977f2fa15294eb56035c43b6

Reply:
for L315, done have this question

Reply:
https://github.com/pingcap/tidb-tools/pull/256/commits/0ad3dcfa1ddf4073164486b5fe0ef0941a0ae94e
and a `columOffset` map in `ChunkRange` to remove a for loop.(only have one loop in `update` before)